domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -592,8 +616,8 @@ constructor(
         }
     }
 
-    override suspend fun setCaptureMode(captureMode: CaptureMode) {
-        currentSettings.update { old ->
+    override suspend fun setCaptureMode(captureMode: CaptureMode): Boolean {
+        return currentSettings.tryUpdate { old ->
```

Comment:
Calling `setCaptureMode` should change other settings besides `captureMode` if the combination isn't supported. So if the `captureMode` argument passed in is SINGLE_STREAM, then it should change the `imageFormat` to `JPEG` rather than failing.

Similarly, if you call `setImageFormat(JPEG_ULTRA_HDR)`, then it should change the `captureMode` in the settings to `MULTI_STREAM`.

Reply:
Thank you, updated. Please refer to the previous reply.